### ISSUE TITLE
Add crossval function that allows for k-fold cross validation of a machine learning model.

### ISSUE THREAD

**Issue by pchest** [quanteda/quanteda.textmodels#22]
The idea would be to create a function `crossval(x, ...)` that takes a machine learning mdoel as an input and allows users to evaluate the model's performance across k splits of en evaluation data set.

**Comment by stefan-mueller** [comment]
The **quanteda.classifiers** package contains the functions [`crossval()`](https://github.com/quanteda/quanteda.classifiers/blob/master/R/crossval.R) and [`performance()`](https://github.com/quanteda/quanteda.classifiers/blob/master/R/performance.R) which allow for straightforward k-fold cross-validation of `textmodel_nb()` and `textmodel_svm()`. While these models are included in **quanteda.textmodels**, functions for cross-validation are missing in the package.

Would it make sense to add these functions to **quanteda.textmodels**? This will allow users to validate their models without having to install the development package **quanteda.classifiers** (which also imports **keras**)?

**Comment by kbenoit** [comment]
I wonder if this is the best approach, or whether an integration into the new(er) tidymodels framework would be the better way to proceed.

**Comment by kbenoit** [comment]
yes exactly - but in a way that extends quanteda.textmodels rather than requiring any new package.

**Comment by pchest** [comment]
Got it. I've cloned `tidymodels` to better understand how their functions work relative to `quanteda`. The functions that seem to be good starting points for improving compatibility are `fit` and `fit_resamples`, as they are crucial to the `tidymodels` k-fold cross-validation workflow.

**Comment by EmilHvitfeldt** [comment]
Hey 👋 just wanted to chime in to say that I'm here to help/answer questions related to any tidymodels effort :)

**Comment by pchest** [comment]
@EmilHvitfeldt Hey! I'm glad to hear you are interested in our project. The objective would be to make `quanteda.textmodel` functions compatible with `tidymodel` model validation functions. For instance, we'd something like the following to work:

```

folds <- vfold_cv(data, v = 10)

nb_mod <- textmodel_nb()
nb_val <- nb_mod %>% 
    fit_resamples(folds)

collect_metrics(nb_val)
```
I'm examining both packages to see what changes would be needed to make this happen. Any suggestions would be welcome!

**Comment by jblumenau** [comment]
Hi all,

Just wondering whether anything came of this idea? I am teaching with quanteda.textmodels this term and had been wondering whether there was a native quanteda cross-validation function for, e.g., textmodel_nb. Cheers!
